sys/include/net/ipv6/addr.h

```diff
@@ -687,11 +690,16 @@ static inline void ipv6_addr_set_all_routers_multicast(ipv6_addr_t *addr, unsign
  */
 static inline void ipv6_addr_set_solicited_nodes(ipv6_addr_t *out, const ipv6_addr_t *in)
 {
-    out->u64[0] = byteorder_htonll(0xff02000000000000);
-    out->u32[2] = byteorder_htonl(1);
-    out->u8[12] = 0xff;
+    static const ipv6_addr_t solicited_node_addr = {
+        .u8 = {
+            0xff, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x01, 0xff, 0x00, 0x00, 0x00
+        }
+    };
+    *out = solicited_node_addr;
```

Comment:
Why not use the already predefined prefix here?

```suggestion
    *out = ipv6_addr_solicited_node_prefix;
```